mlx/coverity_services.py

```diff
@@ -1,253 +1,226 @@
+    def retrieve_issues(self, filters):
+        """Retrieve issues from the server (Coverity Connect).
+
+        Args:
+            filters (json): The filters as json
+
+        Returns:
+            dict: The response
+        """
+        url = self.base_url.rstrip('/') + \
+            "/issues/search?includeColumnLabels=true&offset=0&queryType=bySnapshot&rowCount=-1&sortOrder=asc"
+        return self._post_request(url, filters)
+
+    def retrieve_column_keys(self):
+        """Retrieves the column keys and associated display names.
+
+        Returns:
+            list[dict]: A list of dictionaries where the keys of each dictionary are 'columnKey' and 'name'
+        """
+        if not self._columns:
+            url = f"{self.base_url.rstrip('/')}/issues/columns?queryType=bySnapshot&retrieveGroupByColumns=false"
+            self._columns = self._get_request(url)
+        return self.columns
+
+    def retrieve_checkers(self):
+        """Retrieve the list of checkers from the server.
+
+        Returns:
+            list[str]: The list of valid checkers
+        """
         if not self.checkers:
-            self.checkers = self.client.service.getCheckerNames()
+            url = f"{self.base_url.rstrip('/')}/checkerAttributes/checker"
+            checkers = self._get_request(url)
+            if checkers and "checkerAttributedata" in checkers:
+                self._checkers = [checker["key"] for checker in checkers["checkerAttributedata"]]
         return self.checkers
 
+    def _get_request(self, url):
+        """Make a GET request to the API.
+
+        Args:
+            url (str): The url to request data via GET request
+
+        Returns:
+            dict: the content of server's response
+
+        Raises:
+            requests.HTTPError
+        """
+        req = self.session.get(url)
+        if req.ok:
+            return json.loads(req.content)
+        else:
+            try:
+                message = json.loads(req.content)["message"]
+            except:
+                message = req.content.decode()
+            logger = getLogger("coverity_logging")
+            logger.warning(message)
+            return req.raise_for_status()
+
+    def _post_request(self, url, data):
+        """Perform a POST request to the specified url.
+
+        Args:
+            url (str): The url to request data via POST request
+            data (dict): The data to send
+
+        Returns:
+            dict: the content of server's response
+
+        Raises:
+            requests.HTTPError
+        """
+        req = self.session.post(url, json=data)
+        if req.ok:
+            return json.loads(req.content)
+        else:
+            try:
+                message = json.loads(req.content)["message"]
+            except:
+                message = req.content.decode()
+            logger = getLogger("coverity_logging")
+            logger.warning(message)
+            return req.raise_for_status()
```

Comment:
`_get_request` and `_post_request` share a lot of code. This functionality can easily be combined in a single funciton. One of them expects an extra argument.  Make it an optional keyword argument and this is how you can determine whether to use `self.session.post` or `self.session.get`.

Reply:
Resolved in d912b0619bf193011ae954d80a8207a4ea30e22a and 516e2ab706a8f1a9959ee7b467829c29bded5fd3